src/app.rs

```diff
@@ -130,7 +129,35 @@ where
                     .level(tracing::Level::ERROR),
             ),
     );
-    let tcp_listener = tokio::net::TcpListener::bind(&socket_addr).await?;
-    let server = axum::serve(tcp_listener, router.into_make_service());
-    Ok(server)
+
+    logger::info!(
+        "Locker started [{:?}] [{:?}]",
+        global_app_state.global_config.server,
+        global_app_state.global_config.log
+    );
+
+    logger::debug!(startup_config=?global_app_state.global_config);
+
+    #[cfg(feature = "tls")]
+    {
+        let tcp_listener = std::net::TcpListener::bind(socket_addr)?;
+        let tls_config = RustlsConfig::from_pem_file(
+            &global_app_state.global_config.tls.certificate,
+            &global_app_state.global_config.tls.private_key,
+        )
+        .await?;
+
+        axum_server::from_tcp_rustls(tcp_listener, tls_config)
+            .serve(router.into_make_service())
+            .await?;
+    }
+
+    #[cfg(not(feature = "tls"))]
+    {
+        let tcp_listener = tokio::net::TcpListener::bind(socket_addr).await?;
+
+        axum::serve(tcp_listener, router.into_make_service()).await?;
```

Comment:
can we control this in environment variables instead of compile time feature flag?